examples/nanocoap_server/coap_handler.c

```diff
@@ -182,6 +186,48 @@ NANOCOAP_RESOURCE(sha256) {
     .path = "/sha256", .methods = COAP_POST, .handler = _sha256_handler
 };
 
+/* separate response requires an event thread to execute it */
+#ifdef MODULE_EVENT_THREAD
+static nanocoap_sock_response_ctx_t _separate_ctx;
+static bool _separate_in_progress;
+
+static void _send_response(void *ctx)
+{
+    const char response[] = "This is a delayed response.";
+
+    puts("_separate_handler(): send delayed response");
+    nanocoap_sock_send_separate(ctx, COAP_CODE_CONTENT, COAP_TYPE_NON,
+                                response, sizeof(response));
+    _separate_in_progress = false;
+}
+
+static ssize_t _separate_handler(coap_pkt_t *pkt, uint8_t *buf, size_t len, coap_request_ctx_t *context)
+{
+    static event_timeout_t event_timeout;
+    static event_callback_t event_timed = EVENT_CALLBACK_INIT(_send_response, &_separate_ctx);
+
+    if (_separate_in_progress) {
+        puts("_separate_handler(): response already scheduled");
+        return coap_build_reply(pkt, COAP_CODE_SERVICE_UNAVAILABLE, buf, len, 0);
```

Comment:
```
If a retransmitted request is
 received (perhaps because the original Acknowledgement was delayed),
 another Empty Acknowledgement is sent, and any response MUST be sent
 as a separate response.
```

Reply:
Ah this is for another client who tries to get `/separate`.

Reply:
Yea, I thought for a simple server it should be enough to not support concurrent requests.

Reply:
[4.29](https://datatracker.ietf.org/doc/html/rfc8516), "Too Many Requests" might be better suited though

Reply:
>    The 4.29 response code is only returned to the client(s) sending
   requests too frequently; if other clients are sending requests that
   cannot be served due to server overload, the 5.03 response code is
   more appropriate.

I understand 4.29 as: "You client(s) <X, Y, Z> specifically you need to slow down a little bit". Instead  5.03 sounds like it is the best choice here